presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ExpressionEquivalence.java

```diff
@@ -306,7 +306,7 @@ else if (rightValue == null) {
 
                 // value is some random type (say regex), so we just randomly choose a greater value
```

Comment:
the comment is now incorrect

Reply:
Its not incorrect, 0 means we don't care how java sort will order it...